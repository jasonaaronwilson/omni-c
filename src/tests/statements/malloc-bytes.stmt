malloc_bytes(sizeof(typeof(value_alist_t*)) * n_buckets);
/* result->buckets = cast(typeof(value_alist_t**), malloc_bytes(sizeof(typeof(value_alist_t*)) * n_buckets)); */
/* result->buckets = cast(value_alist_t**, malloc_bytes(sizeof(value_alist_t*) * n_buckets)); */
