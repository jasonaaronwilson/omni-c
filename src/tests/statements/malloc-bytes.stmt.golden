{
    tag = PARSE_NODE_EXPRESSION_STATEMENT
    expression = {
        tag = PARSE_NODE_CALL
        function = {
            tag = PARSE_NODE_IDENTIFIER
            token = malloc_bytes
        }
        args = [
            {
                tag = PARSE_NODE_OPERATOR
                operator = "*"
                left = {
                    tag = PARSE_NODE_OPERATOR
                    operator = sizeof
                    right = {
                        tag = PARSE_NODE_OPERATOR
                        operator = typeof
                        left = {
                            tag = PARSE_NODE_TYPE
                            type_node_kind = TYPE_NODE_KIND_POINTER
                            type_args = [
                                {
                                    tag = PARSE_NODE_TYPE
                                    type_node_kind = TYPE_NODE_KIND_TYPENAME
                                    type_name = value_alist_t
                                    type_args = [
                                    ]
                                }
                            ]
                        }
                    }
                }
                right = {
                    tag = PARSE_NODE_IDENTIFIER
                    token = n_buckets
                }
            }
        ]
    }
}

// C Output
malloc_bytes(((sizeof(typeof(value_alist_t*)))*n_buckets));

